persist/fs/msgpack/decoder.go

```diff
@@ -214,18 +214,24 @@ func (dec *Decoder) DecodeLogMetadata() (schema.LogMetadata, error) {
 	return logMetadata, nil
 }
 
-func (dec *Decoder) decodeIndexInfo() schema.IndexInfo {
-	numFieldsToSkip, ok := dec.checkNumFieldsFor(indexInfoType)
+func (dec *Decoder) decodeIndexInfo(actualVersion int) schema.IndexInfo {
+	numFieldsToSkip, ok := dec.checkNumFieldsFor(indexInfoType, true)
 	if !ok {
 		return emptyIndexInfo
 	}
+
 	var indexInfo schema.IndexInfo
-	indexInfo.Start = dec.decodeVarint()
+	indexInfo.BlockStart = dec.decodeVarint()
 	indexInfo.BlockSize = dec.decodeVarint()
 	indexInfo.Entries = dec.decodeVarint()
 	indexInfo.MajorVersion = dec.decodeVarint()
 	indexInfo.Summaries = dec.decodeIndexSummariesInfo()
 	indexInfo.BloomFilter = dec.decodeIndexBloomFilterInfo()
+
+	if actualVersion > 1 {
+		indexInfo.WrittenAt = dec.decodeVarint()
+		indexInfo.IsSnapshot = dec.decodeBool()
```

Comment:
Hm what if we ever have another type of file like “precompacted” for past or future written data, maybe better to make this field an enum type of what type of fileset it is?

Reply:
yeah good call, I thought the same but didn't do it for some reason

Reply:
done